### PR TITLE
feat(examples): add documentation and walkthroughs for examples

### PR DESCRIPTION
**Pull request by vados-cosmonic**
This commit adds documentation along with walkthroughs for how to use `jco` when building and deploying WebAssembly components.

These examples offer a guide to the components in the repo that mirrors the work in [`component-docs`](https://github.com/bytecodealliance/component-docs), but is slightly easier to access and can contain more specific information w/ closer integration to the example code.

### CODE REVIEW

**Comment by guybedford** [comment]
Looks great to me - it seems like some of the feedback on the docs PR might also apply here (eg referring to a `rev` package), let me know when it's ready for final review?

In the mean time I'm working on the CI issues, should have them green shortly.

**Comment by vados-cosmonic** [comment]
Thanks @guybedford  :bow: will be happy to have my PR be green!

There was a ton of good feed feedback in the docs PR that I missed, so I'm going to maybe wait for that to land and take another look at this. I'll let you know! Converting to draft for now!

File: examples/guides/01-building-a-component-with-jco.md [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+> without having `jco` and `componentize-js` installed globally.
+
+[wit]: https://github.com/WebAssembly/component-model/blob/main/design/mvp/WIT.md
+[nodejs]: https://nodejs.org/en
+[wasi]: https://wasi.dev/
+[wasm-core]: https://webassembly.github.io/spec/core/
+
+## FAQ
+
+### Is this a Reactor component?
```
Comment (vados-cosmonic) [inline examples/guides/01-building-a-component-with-jco.md:65]: Hey @pchickey would love a gutcheck here -- IIRC the difference between command and reactor is the `_start` export and semantics, is this section missing anything?

**Comment by vados-cosmonic** [comment]
Hey @guybedford this should be ready to review! 

In reworking the example tests I found that there's actually been a regression -- I now get this error when running the composed example (it runs in CI on this branch and will run on `main` so we know earlier next time):

```
 wasm://wasm/0288e6d6:1
RuntimeError: unreachable
    at wasm://wasm/0288e6d6:wasm-function[127]:0x24d91
    at wasm://wasm/0288e6d6:wasm-function[128]:0x24d9a
    at wasm://wasm/0288e6d6:wasm-function[4927]:0x224065
    at example:string-reverse-upper/reversed-upper@0.1.0#reverse-and-uppercase (wasm://wasm/0288e6d6:wasm-function[12736]:0x79a5fe)
    at Object.reverseAndUppercase (file:///home/runner/work/jco/jco/examples/components/string-reverse-upper/dist/transpiled/string-reverse-upper.js:58:98)
    at file:///home/runner/work/jco/jco/examples/components/string-reverse-upper/run-transpiled.js:10:30
    ```

**Comment by guybedford** [comment]
Interesting, yeah seems the example is failing. Any ideas?

**Comment by vados-cosmonic** [comment]
:man_facepalming:

```diff
- reverseAndUppercase()
+ reverseAndUppercase(s)
```

**Comment by vados-cosmonic** [comment]
@guybedford should be ready for review now, thanks for fixing those tests -- everything looks :green_circle:  and now the examples should be covered under test going forward.

**Pull request merged**